tests/transforms/test_decompose_transform.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the ``decompose`` transform"""
+
+import pytest
+
+import pennylane as qml
+import pennylane.numpy as qnp
+from pennylane.transforms.decompose import decompose
+
+# pylint: disable=too-few-public-methods
+
+
+class InfiniteOp(qml.operation.Operation):
+    """An op with an infinite decomposition."""
+
+    num_wires = 1
+
+    def decomposition(self):
+        return [InfiniteOp(*self.parameters, self.wires)]
+
+
+class TestDecompose:
+    """Unit tests for decompose function"""
+
+    gate_set_inputs = [None, "RX", ["RX"], ("RX"), {"RX"}, qml.RX, [qml.RX], (qml.RX), {qml.RX}]
```

Comment:
```suggestion
    gate_set_inputs = [None, "RX", ["RX"], ("RX",), {"RX"}, qml.RX, [qml.RX], (qml.RX, ), {qml.RX}]
```
`("RX")` is just the same as `"RX"`, so I'm thinking we were actually intending to test the tuple case here instead.